packages/bootleg-app-contracts/test/BootlegTraderApp.test.js

```diff
@@ -0,0 +1,132 @@
+const BootlegTraderApp = artifacts.require('BootlegTraderApp');
+const BootlegToken = artifacts.require('BootlegToken');
+
+contract('BootlegTraderApp', accounts => {
+  let tokenContract;
+  let app;
+  const tokenId = web3.utils.toBN(1);
+  const mockURI = 'https://ipfs.infura.io/ipfs/QmSomeHash';
+  const firstOwner = accounts[0];
+  const secondOwner = accounts[1];
+  const thirdOwner = accounts[2];
+  const notAnOwner = accounts[9];
+  const oneEthInWei = web3.utils.toWei('1', 'ether');
+  const twoEthInWei = web3.utils.toWei('2', 'ether');
+
+  beforeEach(async () => {
+    tokenContract = await BootlegToken.new('Bootleg', 'BLEG', 5);
+    app = await BootlegTraderApp.new(tokenContract.address, tokenId);
+    await tokenContract.mintWithTokenURI(firstOwner, tokenId, mockURI);
+    await tokenContract.setTraderApp(app.address);
+  });
+
+  it('has a price of zero by default', async () => {
+    expect((await app.getTokenPrice()).toString()).toEqual('0');
+  });
+
+  it('cannot be purchased with a zero price', async () => {
+    let errorMsg = '';
```

Comment:
For future proofing and clarity, might be good to stick a `await app.setPrice(0)` in this test.

Reply:
Good idea